bin/node/src/commands.rs

```diff
@@ -0,0 +1,166 @@
+                },
+                Ok,
+            );
+
+        match key {
+            Ok(k) => {
+                let bytes = k.as_array_ref();
+                Ok(AuraId::from(sr25519::Public::from_raw(*bytes)))
+            }
+            Err(why) => panic!("{}", why),
+        }
+    }
+
+    fn upsert_aleph_key(
+        &self,
+        keystore: &LocalKeystore,
+        key_type: KeyTypeId,
+    ) -> Result<AlephId, Error> {
+        let key = SyncCryptoStore::ed25519_public_keys(&*keystore, key_type)
+            .into_iter()
+            .next()
+            .map_or_else(
+                || {
+                    SyncCryptoStore::ed25519_generate_new(&*keystore, key_type, None)
+                        .map_err(|_| Error::KeyStoreOperation)
+                },
+                Ok,
+            );
+
+        match key {
+            Ok(k) => {
+                let bytes = k.as_array_ref();
+                Ok(AlephId::from(ed25519::Public::from_raw(*bytes)))
+            }
+            Err(why) => panic!("{}", why),
+        }
+    }
+
+    pub fn run(&self, cli: &Cli) -> Result<(), Error> {
+        let chain_id = self.chain_params.chain_id();
+
+        let mut genesis_authorities: Vec<AuthorityKeys> = Vec::new();
+
+        LOCAL_AUTHORITIES.iter().for_each(|authority| {
+            let authority_keystore = self
+                .open_keystore(authority, &chain_id)
+                .unwrap_or_else(|_| panic!("Cannot open keystore for {}", authority));
+
+            let aura_key = self
+                .upsert_aura_key(&authority_keystore, key_types::AURA)
+                .unwrap();
+            let aleph_key = self
+                .upsert_aleph_key(&authority_keystore, aleph_primitives::KEY_TYPE)
+                .unwrap();
+            let account_id = get_account_id_from_seed::<sr25519::Public>(authority);
+
+            // NOTE: we could generate libp2p secrets here and add PeerIds to bootstrap nodes list,
+            // see Substrate's GenerateNodeKeyCmd
+
+            genesis_authorities.push(AuthorityKeys {
+                account_id,
+                aura_key,
+                aleph_key,
+            });
+        });
+
+        info!("Building chain spec");
+
+        let chain_spec = match self.chain_params.chain_id().as_str() {
+            chain_spec::DEVNET_ID => {
+                chain_spec::development_config(self.chain_params.clone(), genesis_authorities)
+            }
+            _ => chain_spec::config(self.chain_params.clone(), genesis_authorities, &chain_id),
+        };
+
+        match chain_spec {
+            Ok(spec) => {
+                let json = sc_service::chain_ops::build_spec(&spec, self.raw)?;
+                if std::io::stdout().write_all(json.as_bytes()).is_err() {
+                    let _ = std::io::stderr().write_all(b"Error writing to stdout\n");
+                }
+            }
+            Err(why) => panic!("{}", why),
```

Comment:
We might as well `.unwrap()` the `chain_spec`

Reply:
Done